### ISSUE TITLE
Check for cycles in DAG

### ISSUE THREAD

**Issue by citibeth** [spack/spack#2498]
Spack should check for cycles in the DAG when doing `spack install` or `spack spec`.  If a cycle is detedted, it should exit gracefully (and display USEFUL information allowing the user to find the cycle).  See #2495

**Comment by alalazo** [comment]
Interesting request also in view of #19501

**Comment by alalazo** [comment]
Closed by #19501, since we have this rule in the new concretizer:
```
% Avoid cycles in the DAG
path(Parent, Child) :- depends_on(Parent, Child).
path(Parent, Descendant) :- path(Parent, A), depends_on(A, Descendant).
:- path(A, B), path(B, A).
```
